httpx_auth/_oauth2/common.py

```diff
@@ -86,3 +87,32 @@ def request_new_grant_with_post(
 class OAuth2:
     token_cache = TokenMemoryCache()
     display = DisplaySettings()
+
+
+class OAuth2BaseAuth(abc.ABC, httpx.Auth):
+    state: Optional[str] = None
+    early_expiry: float
+
+    refresh_token: Optional[Callable]
+
+    def auth_flow(
+        self, request: httpx.Request
+    ) -> Generator[httpx.Request, httpx.Response, None]:
+        token = OAuth2.token_cache.get_token(
+            self.state,
+            early_expiry=self.early_expiry,
+            on_missing_token=self.request_new_token,
+            on_expired_token=(
+                self.refresh_token if "refresh_token" in dir(self) else None
+            ),
+        )
+        self._update_user_request(request, token)
+        yield request
+
+    @abc.abstractmethod
+    def request_new_token(self) -> Union[tuple[str, str], tuple[str, str, int]]:
+        pass  # pragma: no cover
+
+    @abc.abstractmethod
+    def _update_user_request(self, request: httpx.Request, token: str) -> None:
+        pass  # pragma: no cover
```

Comment:
This can be removed and the call replaced by the line done within every sub class.
This will require to move some parameters to this class but they make sense in this context

Reply:
I'm not sure what're you saying ...

Reply:
the implementation is the same in every subclass. Meaning it likely can be bring back here.

Reply:
🤦 of course, didn't noticed it...